core/crypto/randomness/impl/randomness_provider_imp.hpp

```diff
@@ -28,7 +28,7 @@ namespace fc::crypto::randomness {
    private:
     Randomness deriveRandomnessInternal(DomainSeparationTag tag,
                                         Serialization s,
-                                        const ChainEpoch &index);
+                                        int64_t index);
```

Comment:
I prefer to bring ChanEpoch type alias back for readability and maintainability purposes. (just wihtou const &)

Reply:
Either I do, but ChainEpoch is unsigned and this function is supposed to be called with chain_epoch == -1 (do not ask why).

Reply:
tag and index are serialized for seed, and their size should be uint64 and int64.